core/src/main/scala/kafka/log/TimeIndex.scala

```diff
@@ -208,14 +208,15 @@ class TimeIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writable:
   override def sanityCheck() {
     val lastTimestamp = lastEntry.timestamp
     val lastOffset = lastEntry.offset
-    require(_entries == 0 || (lastTimestamp >= timestamp(mmap, 0)),
-      s"Corrupt time index found, time index file (${file.getAbsolutePath}) has non-zero size but the last timestamp " +
-          s"is $lastTimestamp which is no larger than the first timestamp ${timestamp(mmap, 0)}")
-    require(_entries == 0 || lastOffset >= baseOffset,
-      s"Corrupt time index found, time index file (${file.getAbsolutePath}) has non-zero size but the last offset " +
-          s"is $lastOffset which is smaller than the first offset $baseOffset")
-    require(length % entrySize == 0,
-      "Time index file " + file.getAbsolutePath + " is corrupt, found " + length +
-          " bytes which is not positive or not a multiple of 12.")
+    if (_entries != 0 && lastTimestamp < timestamp(mmap, 0))
+      throw new CorruptIndexException(s"Corrupt time index found, time index file (${file.getAbsolutePath}) has " +
+        s"non-zero size but the last timestamp is $lastTimestamp which is no greater than the first timestamp " +
+        s"${timestamp(mmap, 0)}")
+    if (_entries == 0 && lastOffset < baseOffset)
```

Comment:
It seems that _entries == 0 should be _entries != 0?

Reply:
Great catch. Fixed and added a test to verify.